src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7076,6 +7076,16 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
                 CheckReceiverAndRuntimeSupportForSymbolAccess(node, receiver, fieldSymbol, diagnostics);
             }
 
+            // If this is a ref field from another compilation, check for support for ref fields.
+            // No need to check for a reference to a field declared in this compilation since
+            // we check at the declaration site. (Check RefKind after IsFromCompilation() to
+            // avoid cycles for source symbols.
+            if (!fieldSymbol.OriginalDefinition.IsFromCompilation(Compilation) &&
+                fieldSymbol.RefKind != RefKind.None)
```

Comment:
I agree, `fieldSymbol.RefKind == RefKind.None` will be the common case, but the reason to check `IsFromCompilation()` first is to avoid cycles calling `fieldSymbol.RefKind` for fields from source (when binding a group of dependent fields for instance).